xarray/backends/common.py

```diff
@@ -37,6 +57,64 @@ def _normalize_path(path):
     return path
 
 
+def _find_absolute_paths(
+    paths: str | NestedSequence[str | os.PathLike], **kwargs
+) -> list[str]:
+    """
+    Find absolute paths from the pattern.
+
+    Parameters
+    ----------
+    paths :
+        Path(s) to file(s). Can include wildcards like * .
+    **kwargs :
+        Extra kwargs. Mainly for fsspec.
+
+    Examples
+    --------
+    >>> from pathlib import Path
+
+    >>> directory = Path(xr.backends.common.__file__).parent
+    >>> paths = str(Path(directory).joinpath("comm*n.py"))  # Find common with wildcard
+    >>> paths = xr.backends.common._find_absolute_paths(paths)
+    >>> [Path(p).name for p in paths]
+    ['common.py']
+    """
+    if isinstance(paths, str):
+        if is_remote_uri(paths) and kwargs.get("engine", None) == "zarr":
+            try:
+                from fsspec.core import get_fs_token_paths
+            except ImportError as e:
+                raise ImportError(
+                    "The use of remote URLs for opening zarr requires the package fsspec"
+                ) from e
+
+            fs, _, _ = get_fs_token_paths(
+                paths,
+                mode="rb",
+                storage_options=kwargs.get("backend_kwargs", {}).get(
+                    "storage_options", {}
+                ),
+                expand=False,
+            )
+            tmp_paths = fs.glob(fs._strip_protocol(paths))  # finds directories
+            paths = [fs.get_mapper(path) for path in tmp_paths]
+        elif is_remote_uri(paths):
+            raise ValueError(
+                "cannot do wild-card matching for paths that are remote URLs "
+                f"unless engine='zarr' is specified. Got paths: {paths}. "
+                "Instead, supply paths as an explicit list of strings."
+            )
+        else:
+            paths = sorted(glob(_normalize_path(paths)))
+    elif isinstance(paths, os.PathLike):
+        paths = [os.fspath(paths)]
```

Comment:
I know that this is just moved from the previous location, but is there a reason why we don't convert to absolute path here and below?

Reply:
I'm not sure, it's a little odd to me as well. 

I haven't really felt the need for os.PathLike in my normal usage so I don't have much experience with that part though.